src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -47,59 +75,152 @@ private CoseHeaderMap(bool isReadOnly)
 
+        /// <summary>
+        /// Adds the specified key and value to the <see cref="CoseHeaderMap"/>. The specified value will be encoded as CBOR and stored as a <see cref="CoseHeaderValue"/>.
+        /// </summary>
+        /// <param name="label">The key of the element to add.</param>
+        /// <param name="value">The value of the element to encode and then add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
         public void Add(CoseHeaderLabel label, byte[] value) => Add(label, CoseHeaderValue.FromBytes(value));
 
+        /// <summary>
+        /// Adds the specified key and value to the <see cref="CoseHeaderMap"/>. The specified value will be encoded as CBOR and stored as a <see cref="CoseHeaderValue"/>.
+        /// </summary>
+        /// <param name="label">The key of the element to add.</param>
+        /// <param name="value">The value of the element to encode and then add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
         public void Add(CoseHeaderLabel label, ReadOnlySpan<byte> value) => Add(label, CoseHeaderValue.FromBytes(value));
 
+        /// <inheritdoc/>
         public bool ContainsKey(CoseHeaderLabel key) => _headerParameters.ContainsKey(key);
 
+        /// <inheritdoc/>
         public bool TryGetValue(CoseHeaderLabel key, out CoseHeaderValue value) => _headerParameters.TryGetValue(key, out value);
 
+        /// <summary>
+        /// Removes all keys and values from the <see cref="CoseHeaderMap"/>.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
         public void Clear()
         {
             ValidateIsReadOnly();
             _headerParameters.Clear();
         }
 
+        /// <inheritdoc/>
         public bool Contains(KeyValuePair<CoseHeaderLabel, CoseHeaderValue> item)
             => HeaderParametersAsCollection.Contains(item);
 
+        /// <inheritdoc/>
         public void CopyTo(KeyValuePair<CoseHeaderLabel, CoseHeaderValue>[] array, int arrayIndex)
             => HeaderParametersAsCollection.CopyTo(array, arrayIndex);
 
+        /// <inheritdoc/>
         public IEnumerator<KeyValuePair<CoseHeaderLabel, CoseHeaderValue>> GetEnumerator()
             => _headerParameters.GetEnumerator();
 
+        /// <inheritdoc/>
         IEnumerator IEnumerable.GetEnumerator()
             => _headerParameters.GetEnumerator();
 
+        /// <summary>
+        /// Removes the value with the specified label from the <see cref="CoseHeaderMap"/>.
+        /// </summary>
+        /// <param name="label">The label of the element to remove.</param>
+        /// <returns><see langword="true"/> if the element is successfully found and removed; otherwise, <see langword="false"/>. This method returns false if <paramref name="label"/> is not found in the map.</returns>
```

Comment:
The second sentence is largely redundant.  Removing "and removed" from the first makes it completely clear.

```suggestion
        /// <returns><see langword="true"/> if the label was found in the map; otherwise, <see langword="false"/>.</returns>
```

There's no state where the method returns, the label was found, and it wasn't removed.  (That state would throw)